kotlin-inject-compiler/core/src/main/kotlin/me/tatarka/inject/compiler/TypeResultGenerator.kt

```diff
@@ -128,20 +128,29 @@ data class TypeResultGenerator(val options: Options, val implicitAccessor: Acces
             }
 
             if (changeScope) {
-                add("\n")
-                endControlFlow()
+                // don't use endControlFlow() because it emits a newline after the closing brace
+                add("\n⇤}")
             }
         }.build()
     }
 
     private fun TypeResult.Constructor.generate(): CodeBlock {
         return CodeBlock.builder().apply {
             add("%T(", type.asTypeName())
-            parameters.forEachIndexed { i, param ->
+            if (parameters.isNotEmpty()) {
+              val constructorParams = type.toAstClass().primaryConstructor?.parameters ?: emptyList()
```

Comment:
Think it would be better to add a `name` property to `TypeResult.Constructor`, then you don't have to re-look-it-up.

Reply:
To be more specific, you can change the type of `parameters` from `List<TypeResultRef>` to `Map<String, TypeResultRef>`

Reply:
I added a new commit with those changes. Because of that I was also able to provide named parameters to functions.